types/node/index.d.ts

```diff
@@ -5033,7 +5084,7 @@ declare module "stream" {
             isPaused(): boolean;
             unpipe<T extends NodeJS.WritableStream>(destination?: T): this;
             unshift(chunk: any): void;
-            wrap(oldStream: NodeJS.ReadableStream): Readable;
+            wrap(oldStream: NodeJS.ReadableStream): NodeJS.ReadableStream;
```

Comment:
so maybe better is `wrap<T extends NodeJS.ReadableStream>(oldStream: NodeJS.ReadableStream): T`

It will break some modules which define `wrap` method, but it is very easy to fix (maybe 3 modules in DefinitelyTyped repo)

Reply:
I think it should be `wrap(oldStream: NodeJS.ReadableStream): this` as this is how it is implemented.